GLMakie/src/GLAbstraction/GLInfo.jl

```diff
@@ -97,3 +97,12 @@ function getProgramInfo(p::GLProgram)
     @show info = glGetProgramiv(program, GL_TRANSFORM_FEEDBACK_BUFFER_MODE)
     @show info = glGetProgramiv(program, GL_TRANSFORM_FEEDBACK_VARYINGS)
 end
+
+const FAILED_FREE_COUNTER = Ref(0)
+
+function verify_free(obj::T, name = string(T)) where T
+    if obj.id != 0
+        FAILED_FREE_COUNTER[] = FAILED_FREE_COUNTER[] + 1
+        Threads.@spawn println(stderr, "Error: $name has not been freed.")
+    end
+end
```

Comment:
This is used as the finalizer now, to "error" when something hasn't been deleted. I also added a counter to query in CI. We could also de/activate that with an environment variable but I figured incrementing and integer doesn't really matter.

Reply:
TODO: 

`string(T)` seems to be pretty slow, should be moved so it only happens if the branch is true (dragged one of my benchmarks in #4689 from ~300µs down to 800µs)

Reply:
Didn't we want to completely disable finalizers and only activate them in debug mode?

Reply:
I think registering a finalizer is somewhat expensive, so if we only use it for debugging purposes we should just disable it via a global ( not an env var)